openslides/core/projector.py

```diff
@@ -63,50 +66,72 @@ def validate_config(cls, config_data):
         """
         if not isinstance(config_data.get('countdown_time'), (int, float)):
             raise ProjectorException('Invalid countdown time. Use integer or float.')
-        if config_data.get('status') not in ('running', 'stop'):
-            raise ProjectorException("Invalid status. Use 'running' or 'stop'.")
+        if not isinstance(config_data.get('running'), bool):
+            raise ProjectorException("Invalid running status. Has to be a boolean.")
         if config_data.get('default') is not None and not isinstance(config_data.get('default'), int):
             raise ProjectorException('Invalid default value. Use integer.')
 
     @classmethod
-    def control(cls, action, projector_id=1, index=0):
-        """
-        Starts, stops or resets the countdown with the given index on the
-        given projector.
-
-        Action must be 'start', 'stop' or 'reset'.
-        """
+    def control(cls, action):
         if action not in ('start', 'stop', 'reset'):
-            raise ValueError("Action must be 'start', 'stop' or 'reset', not {}.".format(action))
-
-        projector_instance = Projector.objects.get(pk=projector_id)
-        projector_config = {}
-        found = False
-        for key, value in projector_instance.config.items():
-            if value['name'] == cls.name:
-                if index == 0:
-                    try:
-                        cls.validate_config(value)
-                    except ProjectorException:
-                        # Do not proceed if the specific procjector config data is invalid.
-                        # The variable found remains False.
-                        break
-                    found = True
-                    if action == 'start' and value['status'] == 'stop':
-                        value['status'] = 'running'
-                        value['countdown_time'] = now().timestamp() + value['countdown_time']
-                    elif action == 'stop' and value['status'] == 'running':
-                        value['status'] = 'stop'
-                        value['countdown_time'] = value['countdown_time'] - now().timestamp()
-                    elif action == 'reset':
-                        value['status'] = 'stop'
-                        value['countdown_time'] = value.get('default', config['projector_default_countdown'])
-                else:
-                    index += -1
-            projector_config[key] = value
-        if found:
-            projector_instance.config = projector_config
-            projector_instance.save()
+                raise ValueError("Action must be 'start', 'stop' or 'reset', not {}.".format(action))
+
+        if (Countdown_model.objects.all().exists()):
```

Comment:
It looks like that this logic can be moved to a viewset if we decide to use models for countdowns.